### PR TITLE
Implement explicit kinds on type alias parameters

### LINKED ISSUE #221: Allow kinds to be specified on type variables declarations
```
type Monad (m : Type -> Type) = ...
```

### PR DESCRIPTION
**Pull request by brendanzab**
Fixes #221

### CODE REVIEW

File: check/src/typecheck.rs [hunk 1]
```diff
@@ -933,18 +933,24 @@ impl<'a> Typecheck<'a> {
...
                 let mut id_kind = check.type_kind();
                 let alias = Alias::make_mut(&mut bind.alias);
-                for gen in alias.args.iter_mut().rev() {
-                    gen.kind = check.subs.new_var();
-                    id_kind = Kind::function(gen.kind.clone(), id_kind);
+                for generic in alias.args.iter_mut().rev() {
+                    match *generic.kind {
+                        Kind::Hole => generic.kind = check.subs.new_var(),
+                        Kind::Variable(_) => panic!("Unexpected kind variable"),
+                        Kind::Function(_, _) => unimplemented!(),
```
Comment (brendanzab) [inline check/src/typecheck.rs]: Seems like we might have to do some sort of recursive thingy here...

Comment (Marwes) [inline check/src/typecheck.rs]: Yeah, just as type signatures are instantiated.

File: check/tests/pass.rs [hunk 2]
```diff
@@ -866,3 +866,28 @@ make_applicative applicative_Function
...
+    let _ = ::env_logger::init();
+    let text = r#"
+type Test (a : Type -> Type) = a Int
+type Foo a = a
+type Bar = Test Foo
+()
+"#;
+    let result = support::typecheck(text);
+    assert!(result.is_ok(), "{}", result.unwrap_err());
+}
```
Comment (Marwes) [inline check/tests/pass.rs]: Could you add a test with kind holes and a test using `Row` as well?

Comment (brendanzab) [inline check/tests/pass.rs]: Not sure how to implement a `Row` test with out current syntax... :/

Comment (Marwes) [inline check/tests/pass.rs]: ok I am blind

Comment (Marwes) [inline check/tests/pass.rs]: Maybe just something like this

```f#
type Test (r: Row) (f : Row -> Type) = f r
() 
```

Comment (brendanzab) [inline check/tests/pass.rs]: ahhh, good idea!

**Pull request merged**